### ISSUE TITLE
`deno flamegraph`

### ISSUE THREAD

**Issue by KnorpelSenf** [denoland/deno#22102]
In order to analyse application performance, it can be useful to view a flamegraph, much like what `cargo flamegraph` is doing. I suggest to have `deno flamegraph` that outputs a similar SVG file for the JavaScript side.

Deno can already do benchmarks with `deno bench` and perhaps someone has a good idea how to integrate these two related features?

**Comment by mmastrac** [comment]
We may be able to better integrate the v8 profiler to remove a lot of the required steps to get a profiling trace.

**Comment by bartlomieju** [comment]
For starters, as Matt pointed out we need to be able to acquire the profiling data from V8.

To do that we need to write out structures for types listed here: https://chromedevtools.github.io/devtools-protocol/tot/Profiler/#type-Profile This should be done in https://github.com/denoland/deno/blob/main/cli/cdp.rs. 

Once we have that, it should be easy to collect a profile when a flag/env var is specified. Then the real fun begins where we need to provide a visualization for that data.

**Comment by KnorpelSenf** [comment]
> Then the real fun begins where we need to provide a visualization for that data.

@bartlomieju would you want to deviate from `cargo flamegraph` in any way?

**Comment by bartlomieju** [comment]
@KnorpelSenf I don't have strong opinions on this one.

**Comment by KnorpelSenf** [comment]
@bartlomieju does the OTEL setup provide this information now?

**Comment by bartlomieju** [comment]
@KnorpelSenf no, the way to pull out this data is listed [in the comment above](https://github.com/denoland/deno/issues/22102#issuecomment-1910394320).

**Comment by bartlomieju** [comment]
Done in https://github.com/denoland/deno/pull/32572
